CMakeLists.txt

```diff
@@ -0,0 +1,34 @@
+cmake_minimum_required(VERSION 3.4)
+project(coz C CXX)
+
+enable_testing()
+
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+list(APPEND CMAKE_MODULE_PATH ${PROJECT_BINARY_DIR})
+
+include(GNUInstallDirs)
+find_package(Threads REQUIRED)
+find_package(libelfin REQUIRED)
+
+file(GLOB_RECURSE sources libcoz/*.cpp)
+file(GLOB_RECURSE headers include/*.h)
+
+install(PROGRAMS coz DESTINATION bin)
```

Comment:
Install option is done:
https://github.com/plasma-umass/coz/pull/178/commits/c3876783157eac4dfe220303da0095d6cdec65b1

I don't have experience with CPack, but I understand the idea about components. I would appreciate if you can make a separate merge-request later as a demo how this can be done properly. Since it's an optimization, this can be done later once we merge this request.